.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,121 @@
+name: Build and upload to PyPI
+
+on:
+  push:
+    tags:
+      - "*"
+  release:
+    types:
+      - published
+
+concurrency:
+  group: "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}"
+  cancel-in-progress: true
+
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+
+jobs:
+  build_wheels:
+    if: "github.repository == 'MDAnalysis/pytng'"
+    name: Build wheels
+    runs-on: ${{ matrix.buildplat[0] }}
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        buildplat:
+          - [ubuntu-22.04, manylinux_x86_64, x86_64]
+          - [macos-11, macosx_*, x86_64]
+          - [windows-2019, win_amd64, AMD64]
+        python: ["cp39", "cp310", "cp311", "cp312"]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.16.2
+        env:
+          CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
+          CIBW_ARCHS: ${{ matrix.buildplat[2] }}
+          CIBW_BUILD_VERBOSITY: 1
+
+      - name: upload artifacts
+        if: |
+          (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')) ||
+          (github.event_name == 'release' && github.event.action == 'published')
+        uses: actions/upload-artifact@v3
+        with:
+          path: wheelhouse/*.whl
+          retention-days: 7
+
+  build_sdist:
+    if: "github.repository == 'MDAnalysis/pytng'"
+    name: build package source distribution
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build sdist
+        run: pipx run build --sdist
+
+      - name: upload artifacts
+        if: |
+          (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')) ||
+          (github.event_name == 'release' && github.event.action == 'published')
+        uses: actions/upload-artifact@v3
+        with:
+          path: dist/*.tar.gz
+          retention-days: 7
+
+  upload_testpypi:
+    if: |
+      github.repository == 'MDAnalysis/pytng' &&
+      (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/'))
+    name: testpypi upload
+    environment:
```

Comment:
Ok done - this was missing.

I also removed the post upload tests because pytng doesn't package test files.